File: frontend/src/types/history.ts
export interface TxHistoryItem {
  tx_hash: string;
  timestamp: number;
  tx_type: string;
  pool_id: string;
  sender: string;
  amount0: string;
  amount1: string;
  usd_value: number;
  status?: "pending" | "confirmed" | "failed";
}

export interface TxHistoryResponse {
  items: TxHistoryItem[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface WsTxEvent {
  type: "tx";
  tx_hash: string;
  timestamp: number;
  tx_type: string;
  pool_id: string;
  sender: string;
  amount0: string;
  amount1: string;
  usd_value: number;
}

export interface LocalTx {
  tx_hash: string;
  tx_type: string;
  sender: string;
  timestamp: number;
  status: "pending" | "confirmed" | "failed";
}

File: backend/internal/v2/dto/history.go
package dto

type TxHistoryItem struct {
	TxHash    string  `json:"tx_hash"`
	Timestamp uint32  `json:"timestamp"`
	TxType    string  `json:"tx_type"`
	PoolID    string  `json:"pool_id"`
	Sender    string  `json:"sender"`
	Amount0   string  `json:"amount0"`
	Amount1   string  `json:"amount1"`
	USDValue  float64 `json:"usd_value"`
}

type TxHistoryResponse struct {
	Items      []TxHistoryItem `json:"items"`
	Total      int             `json:"total"`
	Page       int             `json:"page"`
	PageSize   int             `json:"page_size"`
	TotalPages int             `json:"total_pages"`
}
